docs/guide/design_engine.md

```diff
@@ -116,132 +111,85 @@ For assets, a special **`ezdesign` asset package** is available.
-
-Calling `asset("js/foo.js", "ezdesign")` can for example be resolved to `web/bundles/app/themes/my_theme/js/foo.js`.
-
-### Performance and asset resolution
 
-When using themes, paths for assets are resolved at runtime.
-This is due to how the Symfony Asset component is integrated with Twig.
-This can cause significant performance impact because of I/O calls when looping over all potential theme directories,
-especially when using a lot of different designs and themes.
+### Fallback order
 
-To work around this issue, assets resolution can be provisioned at compilation time.
-Provisioning is the **default behavior in non-debug mode** (e.g. `prod` environment).
-In debug mode (e.g. `dev` environment), assets are being resolved at runtime.
+The default fallback order for templates is:
 
-This behavior can, however, be controlled by the `disable_assets_pre_resolution` setting.
+- Application theme directory: `templates/themes/<theme_name>`
+- Bundle theme directory: `src/<bundle_directory>/Resources/views/themes/<theme_name>`
 
-```yaml
-# ezplatform_prod.yaml
-ezdesign:
-    # Force runtime resolution
-    # Default value is '%kernel.debug%'
-    disable_assets_pre_resolution: true
-```
+The default fallback order for assets is:
 
-## Design usage with templates
+- Application assets directory: `assets`
+- Application theme directory: `assets/themes/<theme_name>`
+- Bundle theme directory: `public/bundles/<bundle_directory>/themes/<theme_name>`
 
-By convention, a theme directory must be located under `<bundle_directory>/Resources/views/themes/` or global
-`templates/themes/` directories.
+Calling `asset("js/scripts.js", "ezdesign")` can for example be resolved to `public/bundles/app/themes/my_theme/js/scripts.js`.
 
-Typical paths can be for example:
-- `templates/themes/foo/` => Templates will be part of the `foo` theme.
-- `templates/themes/bar/` => Templates will be part of the `bar` theme.
-- `src/Acme/TestBundle/Resources/views/themes/the_best/` => Templates will be part of `the_best` theme.
+Bundles fall back in the instantiation order from `bundles.php`.
 
-In order to use the configured design with templates, you need to use the **`@ezdesign`** special **Twig namespace**.
+#### Additional theme paths
 
-```html+twig
-{# Will load 'some_template.html.twig' directly under one of the specified theme directories #}
-{{ include("@ezdesign/some_template.html.twig") }}
+You can also add any Twig template folder to the theme configuration.
 
-{# Will load 'another_template.html.twig', located under 'full/' directory, which is located under one of the specified theme directories #}
-{{ include("@ezdesign/full/another_template.html.twig") }}
-```
+You can use it if you want to define templates from third-party bundles as part of one of your theme,
+or when upgrading your application in order to use the design engine,
+if your existing templates are not yet following the convention.
```

Comment:
```suggestion
and if your existing templates are not yet following the convention.
```
It's actually upgrading+existing template not following convention, so maybe this way to clarify it's not A or B or C.